HTTP_server.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+import email.utils
+import os
+
+
+def server_socket_function():
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+
+            recieve_total = ""
+            buffersize = 32
+            finished = 0
+            while not finished:
+                recieve = conn.recv(buffersize)
+                if len(recieve) < buffersize:
+                    recieve_total += recieve
+                    finished = 1
+                else:
+                    recieve_total += recieve
+
+            if recieve_total:
+                try:
+                    uri = parse_request(recieve_total)
+                    info = resolve_uri(uri)
+                    response = response_ok(info)
+                except RequestError, msg:
+                    errors = str(msg).strip("'").split(' ', 1)
+                    response = response_error(errors[0], errors[1])
+                conn.sendall(response)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def response_ok(*args):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: {}'.format(args[0][0])
+    body = '''<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n</head>\n<body>\n{}</body>\n</html>'''.format(args[0][1])
```

Comment:
The entire purpose of triple-quoted strings is to allow you to break lines.  You can make this look a lot neater by doing something like this:

```python

HTML_PAGE = """
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
  </head>
  <body>
{}
  </body>
</html>
"""

def response_ok(*args):
    # ...
    body = HTML_PAGE.format(args[0][1])
    #...
```

You can even share the `HTML_PAGE` constant you've defined with the `response_error` function below, since it's the same thing.  

You get pep8 wins, DRY code wins and easier code to read all in one.